### PR TITLE
Some basic projectile stuff

### PR DESCRIPTION
**Pull request by Couls**
The projectiles position and effects will line up with the person being hit, also removes a lot of bump code and lets BYOND handle it, adds basic bounds to projectiles

### CODE REVIEW

File: code/modules/projectiles/projectile.dm [hunk 1]
```diff
@@ -11,6 +11,10 @@
 	pass_flags = PASSTABLE
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
 	movement_type = FLYING
+	bound_height = 16
+	bound_width = 16
+	bound_x = 8
+	step_size = 16 // half a tile zoomer
```
Comment (Couls) [inline code/modules/projectiles/projectile.dm:17]: Probably should remove this line and keep them tile based until projectiles are refactored entirely

File: code/__HELPERS/movement.dm [hunk 2]
```diff
@@ -13,14 +13,16 @@
...
-	y += dist * sin(deg)
+	x += dist * sin(deg)
+	y += dist * cos(deg)
 	NORMALIZE_STEP(place, x, y)
 	return thing.Move(place, get_dir(thing.loc, place), x, y)
 
 // Returns the direction from thingA to thingB in degrees
 // EAST is 0 and goes counter clockwise
 /proc/get_deg(atom/movable/thingA, atom/movable/thingB)
+	if(QDELETED(thingA) || QDELETED(thingB))
```
Comment (ninjanomnom) [inline code/__HELPERS/movement.dm]: let it break

File: code/__HELPERS/movement.dm [hunk 3]
```diff
@@ -6,19 +8,33 @@
 
 // Like step but you move on an angle instead of a cardinal direction
 /proc/degstep(atom/movable/thing, deg, dist)
+	if(!thing)
```
Comment (ninjanomnom) [inline code/__HELPERS/movement.dm]: let it break

File: code/modules/projectiles/projectile.dm [hunk 4]
```diff
@@ -355,7 +342,7 @@
 			time_offset += overrun * speed
 		time_offset += MODULUS(elapsed_time_deciseconds, speed)
 
-	for(var/i in 1 to required_moves)
+	for(var/i in 1 to required_moves * 2) // if required moves is turfs and we move 16 pixels per pixel_move we need to double this
```
Comment (ninjanomnom) [inline code/modules/projectiles/projectile.dm]:
```suggestion
	for(var/i in 1 to required_moves * (32 / step_size)) // required moves is in turfs
```

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Pull request merged**